urllib3/poolmanager.py

```diff
@@ -129,22 +174,32 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # Return False to re-raise any potential exceptions
         return False
 
-    def _new_pool(self, scheme, host, port):
+    def _new_pool(self, scheme, host, port, request_context=None):
         """
-        Create a new :class:`ConnectionPool` based on host, port and scheme.
+        Create a new :class:`ConnectionPool` based on host, port, scheme, and
+        any additional pool keyword arguments.
 
-        This method is used to actually create the connection pools handed out
-        by :meth:`connection_from_url` and companion methods. It is intended
-        to be overridden for customization.
+        If `request_context` is provided, it is provided as keyword arguments
```

Comment:
Nit: We seem to be using double ticks (\`\`) in all the other docstrings in this file. May want to move \`request_context\` inline with that.